From f152711d14d2f1ea3f7106463270eb9ebc76c794 Mon Sep 17 00:00:00 2001
From: Patrick Boettcher <patrick.boettcher@posteo.de>
Date: Fri, 20 Nov 2015 15:09:36 +0100
Subject: [PATCH 219/222] SDIO-overlay: add poll_once-boolean parameter

Add paramter to toggle sdio-device-polling
done every second or once at boot-time.

Signed-off-by: Patrick Boettcher <patrick.boettcher@posteo.de>
---
 arch/arm/boot/dts/overlays/README           | 3 +++
 arch/arm/boot/dts/overlays/sdio-overlay.dts | 6 +++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -610,6 +610,9 @@ Params: overclock_50             Clock (
 
         debug                    Enable debug output (default off)
 
+        poll_once                Disable SDIO-device polling every second
+                                 (default on: polling once at boot-time)
+
 
 Name:   smi
 Info:   Enables the Secondary Memory Interface peripheral. Uses GPIOs 2-25!
--- a/arch/arm/boot/dts/overlays/sdio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdio-overlay.dts
@@ -7,7 +7,7 @@
 
 	fragment@3 {
 		target = <&mmc>;
-		__overlay__ {
+		sdio_mmc: __overlay__ {
 			compatible = "brcm,bcm2835-mmc";
 			pinctrl-names = "default";
 			pinctrl-0 = <&sdio_pins>;
@@ -26,4 +26,8 @@
 			};
 		};
 	};
+
+	__overrides__ {
+		poll_once = <&sdio_mmc>,"non-removable?";
+	};
 };
